pg_stat_kcache.c

```diff
@@ -726,7 +734,7 @@ pgsk_hash_fn(const void *key, Size keysize)
 
 	return hash_uint32((uint32) k->userid) ^
 		hash_uint32((uint32) k->dbid) ^
-		hash_uint32((uint32) k->queryid);
+		hash_uint32((t_queryid) k->queryid);
```

Comment:
BTW, hash_uint32 return a Datum, not an uint32.  We should do as what pgss is now doing: use tag_hash as hash function instead of defining pgsk_hash_fn.  pgss doesn"t specify it since it's default behavior since 9.5 when HASH_BLOB flag is set, but pgsk supports older realases.

Reply:
Thanks, I'll redo this part as inspired from pgss.

Reply:
After second thought, it's best to not switch to tag_hash, there's too much problem with this.

However, hash_uint32() expect a uint32, so I think here you should cast k->queryid to uint32, not t_queryid.  Otherwise, maybe rename t_queryid to pgsk_query to avoid any trouble in debugger.  Otherwise I'm happy with it :)